ui/ui-components/components/record/SampleRecordCard.tsx

```diff
@@ -0,0 +1,357 @@
+import Link from "next/link";
+import { Fragment, useCallback, useEffect, useMemo, useState } from "react";
+import { Button, Card, Col, Row, Table } from "react-bootstrap";
+import EnterpriseLink from "../../components/GrouparooLink";
+import { useGrouparooModelContext } from "../../contexts/grouparooModel";
+import { ApiHook } from "../../hooks/useApi";
+import { usePrevious } from "../../hooks/usePrevious";
+import { Actions, Models } from "../../utils/apiData";
+import { successHandler } from "../../utils/eventHandlers";
+import { grouparooUiEdition } from "../../utils/uiEdition";
+import StateBadge from "../badges/StateBadge";
+import ManagedCard from "../lib/ManagedCard";
+import SeparatedItems from "../lib/SeparatedItems";
+import LinkButton from "../LinkButton";
+import LoadingButton from "../LoadingButton";
+import AddSampleRecordModal from "./AddSampleRecordModal";
+import ArrayRecordPropertyList from "./ArrayRecordPropertyList";
+
+interface Props {
+  execApi: ApiHook["execApi"];
+  properties: Models.PropertyType[];
+  disabled: boolean;
+}
+
+const isConfigUI = grouparooUiEdition() === "config";
+
+const getCachedSampleRecordId = (modelId: string): string => {
+  return globalThis.localStorage?.getItem(`sampleRecord:${modelId}`);
+};
+
+const setCachedSampleRecordId = (modelId: string, recordId: string): void => {
+  recordId &&
+    globalThis.localStorage?.setItem(`sampleRecord:${modelId}`, recordId);
+};
+
+const clearCachedSampleRecordId = (modelId: string): void => {
```

Comment:
Not sure how helpful the void return types are _here_. I know this is something we want to discuss!

Reply:
This is a really simple funciton but in general it guarantees that the implementation complies with the definition of the function instead of leaving it to the implementation to define the return type.